core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -198,9 +198,23 @@ public boolean isReady() {
   /**
    * Enables per-message compression, if an encoding type has been negotiated.  If no message
    * encoding has been negotiated, this is a no-op.
+   *
+   * @param enabled whether or not subsequent calls to {@link #sendMessage} should be compressed.
    */
   @ExperimentalApi
   public void setMessageCompression(boolean enabled) {
-    // noop
+    throw new UnsupportedOperationException("unimplemented");
```

Comment:
I don't really accept meh. Defining what happens when compression is not supported is part of the API. With the `UnsupportedOperationException` in place here today, the example for using this API would need to handle the `UnsupportedOperationException`. If you expect it to be noops when unsupported, make it a noop here. If it is supposed to be an exception when unsupported, keep the exception here and make a note of it in the JavaDoc.